tests/test_stack.py

```diff
@@ -222,3 +222,28 @@ def test_tag_floating_state_on(hlwm, focus_idx):
     # we didn't do any raising since
     hlwm.call('floating off')
     assert helper_get_stack_as_list(hlwm) == stack_before
+
+
+def test_focused_on_other_monitor_above_fullscreen(hlwm):
+    # on the unfocused monitor, put a fullscreen window
+    # and focus another window. then the focused window should
+    # be visible!
```

Comment:
I would add a "after switching monitor focus" somewhere in here, this is kind of crucial to this test, right?

Reply:
it is only crucial that it's not on the 'focused' monitor, but it does not matter whether (a) the monitor focus is switched or (b) the other monitor was never focused.  (I just chose (a) here because `jumpto` makes the test easier to write)

Reply:
I've refined the test case to also include (b). On master, both test cases (a and b) fail, and the present PR fixes both :)

Reply:
Thanks for extending the tests! :)